packages/tools/src/apm/install.coffee.md

```diff
@@ -3,73 +3,69 @@
 
 Install Atom packages with APM.
 
-## Options
+## Hooks
 
-*   `name` (string|array)
-    Name of the package(s).
-*   `upgrade` (boolean)
-    Upgrade all packages, default to "false".
+    on_action = ({config}) ->
+      throw Error 'Deprecated config `argument`' if config.argument?
+      config.name = [config.name] if typeof config.name is 'string'
 
 ## Schema
 
     schema =
       type: 'object'
       properties:
-        '':
-          type: 'object'
+        'name':
+          type: 'array', items: type: 'string'
+          description: """
+          Name of the package(s) to install.
+          """
+        'upgrade':
+          type: 'boolean'
+          default: false
           description: """
+          Upgrade all packages.
           """
 
 ## Handler
 
-    handler = ({config}) ->
-      config.name = config.argument if config.argument?
-      config.name = [config.name] if typeof config.name is 'string'
+    handler = ({config, tools: {log}}) ->
       config.name = config.name.map (pkg) -> pkg.toLowerCase()
       outdated = []
       installed = []
       # Note, cant see a difference between update and upgrade after printing help
-      @execute
+      {stdout} = await @execute
         cmd: "apm outdated --json"
         shy: true
-      , (err, {stdout}) ->
-        throw err if err
+      if stdout
```

Comment:
I had `stdout` empty when was running some tests (I don't remember where exactly). So, it might depend on the versions of apm and npm where I've put this.

Reply:
it prints `[]` if no module is outdated. If you experience sth like this, it is required to reproduce it with a test. At the bare minimum shall be a detailed comment. Pretty sure you got empty stdout because there was err error which you are not catching because you left the `shy: true` property. Shy need to be removed everywhere or no error is thrown.

Reply:
Don't you confuse `relax` and `shy`?

Reply:
Yes, this was because of some error. I've fixed